<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Methods</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.2.1/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.2.1/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Risk of mitochondrial deletions is affected by the global secondary structure of the human mitochondrial genome</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">HOME</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Figures</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="Revision_Clustering_mt_deletions.html">Fig.2A. Clustering analysis of mtDNA deletions</a>
        </li>
        <li>
          <a href="figure_2b.html">Fig.2B. Influence of repeats on mtDNA deletions distributon</a>
        </li>
        <li>
          <a href="figure_2cd.html">Fig.2C-D. *In silico* estimates of mtDNA global structure</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="CopelandDeletionsPca.html">Reanalysis of Deletions PCA data from Copeland et al</a>
    </li>
    <li>
      <a href="MitoBreakDeletionsAndOrlovRepeats.html">Exploratory MITOBREAK and Orlovs data analysis</a>
    </li>
    <li>
      <a href="MitoBreak_vs_RandomInMajorArc.html">Compare Good Mitobreak with Rand version of major arc</a>
    </li>
  </ul>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/mitoclub/GlobalStructure">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Methods</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-04-05
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>GlobalStructure/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230404code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230404)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230404code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230404)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommitoclubGlobalStructuretreebaf2fd420af75e5168da862f783d48ecaa29dc66targetblankbaf2fd4a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mitoclub/GlobalStructure/tree/baf2fd420af75e5168da862f783d48ecaa29dc66" target="_blank">baf2fd4</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommitoclubGlobalStructuretreebaf2fd420af75e5168da862f783d48ecaa29dc66targetblankbaf2fd4a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mitoclub/GlobalStructure/tree/baf2fd420af75e5168da862f783d48ecaa29dc66" target="_blank">baf2fd4</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    1_Raw/.DS_Store
    Ignored:    2_Derived/.DS_Store
    Ignored:    3_Results/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/staging/

Unstaged changes:
    Modified:   3_Results/SlipAndJump.HomologyAndRepeats.txt
    Modified:   3_Results/heatmap_folding_stability.svg
    Modified:   3_Results/heatmap_global_folding_sw100.pdf
    Modified:   3_Results/heatmap_microhomology_AIC.pdf
    Modified:   3_Results/heatmap_microhomology_AIC_wt_deledions.svg
    Modified:   3_Results/violin_rep_center_release.pdf
    Modified:   3_Results/violin_rep_flanked_length_release.pdf
    Modified:   3_Results/violin_rep_folding_infsign_np.pdf
    Modified:   3_Results/violin_rep_folding_infsign_p.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/methods.Rmd</code>) and HTML
(<code>docs/methods.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mitoclub/GlobalStructure/blob/baf2fd420af75e5168da862f783d48ecaa29dc66/analysis/methods.Rmd" target="_blank">baf2fd4</a>
</td>
<td>
Evgenii O. Tretiakov
</td>
<td>
2023-04-05
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code># Presentation
library(&quot;glue&quot;)
library(&quot;knitr&quot;)

# JSON
library(&quot;jsonlite&quot;)

# Tidyverse
library(&quot;tidyverse&quot;)</code></pre>
<pre class="r"><code>dir.create(here::here(&quot;3_Results&quot;, DOCNAME), showWarnings = FALSE)

write_bib(c(&quot;base&quot;, &quot;reshape2&quot;, &quot;tidyverse&quot;, &quot;here&quot;, &quot;furrr&quot;, &quot;ggpubr&quot;,
            &quot;cowplot&quot;, &quot;patchwork&quot;, &quot;raster&quot;, &quot;skimr&quot;, &quot;pspearman&quot;,
            &quot;ggstatsplot&quot;, &quot;ggasym&quot;, &quot;cowplot&quot;, &quot;ggstatsplot&quot;,
            &quot;gridExtra&quot;, &quot;tidyverse&quot;, &quot;dplyr&quot;, &quot;tidyr&quot;, &quot;magrittr&quot;, &quot;stringr&quot;, 
            &quot;future&quot;, &quot;purrr&quot;, &quot;here&quot;, &quot;workflowr&quot;, &quot;knitr&quot;,
            &quot;kableExtra&quot;, &quot;rmarkdown&quot;),
          file = here::here(&quot;3_Results&quot;, DOCNAME, &quot;packages.bib&quot;))</code></pre>
<pre class="r"><code>versions &lt;- list(
    cowplot     = packageVersion(&quot;cowplot&quot;),
    dplyr       = packageVersion(&quot;dplyr&quot;),
    furrr       = packageVersion(&quot;furrr&quot;),
    future      = packageVersion(&quot;future&quot;),
    ggasym      = packageVersion(&quot;ggasym&quot;),
    ggpubr      = packageVersion(&quot;ggpubr&quot;),
    ggstatsplot = packageVersion(&quot;ggstatsplot&quot;),
    gridExtra   = packageVersion(&quot;gridExtra&quot;),
    here        = packageVersion(&quot;here&quot;),
    kableExtra  = packageVersion(&quot;kableExtra&quot;),
    knitr       = packageVersion(&quot;knitr&quot;),
    magrittr    = packageVersion(&quot;magrittr&quot;),
    patchwork   = packageVersion(&quot;patchwork&quot;),
    pspearman   = packageVersion(&quot;pspearman&quot;),
    purrr       = packageVersion(&quot;purrr&quot;),
    python      = &quot;3.8.8&quot;,
    R           = str_extract(R.version.string, &quot;[0-9\\.]+&quot;),
    raster      = packageVersion(&quot;raster&quot;),
    reshape2    = packageVersion(&quot;reshape2&quot;),
    rmarkdown   = packageVersion(&quot;rmarkdown&quot;),
    skimr       = packageVersion(&quot;skimr&quot;),
    stringr     = packageVersion(&quot;stringr&quot;),
    tidyr       = packageVersion(&quot;tidyr&quot;),
    tidyverse   = packageVersion(&quot;tidyverse&quot;),
    viridis     = packageVersion(&quot;viridis&quot;),
    workflowr   = packageVersion(&quot;workflowr&quot;)
)</code></pre>
<div id="distribution-of-the-centers" class="section level2">
<h2>Distribution of the centers:</h2>
<p>For each deletion from MitoBreak in the major arc (5781-16569), its
midpoint was found. Next, each of the real deletions was moved randomly
within the major arc, and their midpoints were also obtained. For the
observed means of the observed deletions and randomly simulated ones,
the corresponding standard deviations were obtained and compared.</p>
</div>
<div id="hi-c-mtdna-contact-matrix" class="section level2">
<h2>Hi-C mtDNA contact matrix:</h2>
<p>The publicly available mtDNA matrix was visualized using <a
href="http://aidenlab.org/juicebox/?juiceboxURL=http://bit.ly/2Rmz4wy">Juicebox</a>
<span class="citation">(Robinson et al. 2018)</span>. The corresponding
paper describing the methodology of obtaining Hi-C data derived from the
human lymphoblastoid cell line is by <span class="citation">(Rao et al.
2014)</span>. Additionally, we obtained six Hi-C mtDNA contact matrixes
from olfactory receptors of covid patients and controls. Details of the
in situ Hi-C protocol, as well as bioinformatics analyses, are described
in the original paper <span class="citation">(Zazhytska et al.
2022)</span>. Matrices were visualized using Juicebox <span
class="citation">(Robinson et al. 2018)</span>.</p>
</div>
<div id="in-silico-folding" class="section level2">
<h2>In silico folding:</h2>
<p>We used the heavy chain of the reference human mtDNA sequence
(NC_012920.1) since it spends the most time being single-stranded
according to the asymmetric model of mtDNA replication <span
class="citation">(Persson et al. 2019)</span>. Using Mfold <span
class="citation">(Zuker 2003)</span> with parameters set for DNA folding
and a circular sequence, we constrained everything but the major arc
from forming base pairs. We obtained the global (genome-wide) secondary
structure, which we then translated into the number of hydrogen bonds
connecting our regions of interest (100 bp windows for the analyses and
visualization). Next, within the single-stranded heavy chain of the
major arc, we defined 100 bp windows and hybridized all potential pairs
of such windows using ViennaRna Package 2 <span class="citation">(Lorenz
et al. 2011)</span>. Obtained Gibbs Energies for each pair of such
windows was used as a metric of a strength of a potential interaction
between two single-stranded DNA regions.</p>
</div>
<div id="the-density-of-inverteddirect-repeats" class="section level2">
<h2>The density of inverted/direct repeats:</h2>
<p>For each pair of 100 bp window, we estimated the number of
nucleotides involved in at least one inverted/direct degraded repeat.
The corresponding repeat should have one arm located in the first window
and another arm located in the second window. All degraded (with the
maximal level of imperfection of 80%) repeats in the human mtDNA were
called using our algorithm described previously <span
class="citation">(Shamanskiy et al. 2019)</span>.</p>
</div>
<div id="clusterization-of-deletions" class="section level2">
<h2>Clusterization of deletions:</h2>
<p>For clusterization, we used all MitoBreak <span
class="citation">(Damas et al. 2014)</span> deletions from the major
arc. We used 5’ and 3’coordinates as input for a hierarchical
density-based clustering algorithm (python hdbscan v0.8.24) <span
class="citation">(McInnes and Healy 2017)</span>. DBSCAN is a well-known
algorithm for probability density-based clusterization, which detects
clusters as regions with more densely located sample data as well as
outlier samples. The advantage of this method is soft clustering. We
variated cluster density parameters in order to ensure cluster stability
and found that cluster formations stay relatively stable for a wide
range of parameters. Thus, DBSCAN produces a robust set of clusters,
producing additional evidence for regions with elevated deletion rates.
We also performed affinity propagation clustering <span
class="citation">(Frey and Dueck 2007)</span> as a data exploration
experiment, which also yields robust clustering.</p>
</div>
<div id="perfect-direct-repeats-of-the-human-mtdna"
class="section level2">
<h2>Perfect direct repeats of the human mtDNA:</h2>
<p>The list of the perfect direct repeats with a length of 10 or more
base pairs was used from our algorithm described in <span
class="citation">(Guo et al. 2010)</span>.</p>
</div>
<div id="realized-and-non-realized-direct-degraded-repeats"
class="section level2">
<h2>Realized and non-realized direct degraded repeats:</h2>
<p>We used our database of degraded mtDNA repeats <span
class="citation">(Shamanskiy et al. 2019)</span> with a length of 10 bp
or more and a similarity of 80% or more. We took into account only
direct repeats with both arms located in the major arc. We grouped
repeats with similar motifs into clusters so that each considered
cluster should contain at least three arms of the repeat, and at least
one deletion should be associated with two of them. We additionally
restricted our subset of clusters, considering only non-realized repeats
as pairs of arms, where at least one of them (the first or the second)
is the same as in realized repeat. Visually in Fig 2D, it means that
within each cluster, we compare realized repeats (red dot) with
non-realized ones (grey dot) located on the same horizontal (the same Y
coordinate) or vertical (the same X coordinate) axis. We got 618
clusters like this. Pairwise alignments for microhomology matrix: A
measure for the degree of similarity between segments of the major arc
was obtained by aligning small windows of the mitochondrial major arc
sequence with each other. We sliced the mitochondrial major arc sequence
into 100 nucleotide pieces and aligned them against each other using
EMBOSS Needle <span class="citation">(Needleman and Wunsch 1970)</span>
with default parameters (match +5, gap open - 10, gap extend - 0.5),
parsed out the alignment scores, thus obtaining data for the matrix of
microhomology.</p>
</div>
<div id="other-packages" class="section level1">
<h1>Other packages</h1>
<p>Visualisations and figures were primarily created using the ggplot2
(v), cowplot (v1.1.1) <span class="citation">(Wilke 2020)</span>,
patchwork (v1.1.2) <span class="citation">(Pedersen 2022)</span>, ggasym
(v0.1.6) <span class="citation">(Cook 2021)</span>, ggpubr (v0.6.0)
<span class="citation">(Kassambara 2023)</span> and ggstatsplot
(v0.11.0) <span class="citation">(<strong>patil2021?</strong>)</span>
packages using the viridis colour palette (v0.6.2) for continuous data.
Data manipulation was performed using other packages in the tidyverse
(v2.0.0) <span class="citation">(Wickham 2023)</span> particularly dplyr
(v1.1.1) <span class="citation">(Wickham et al. 2023)</span>, tidyr
(v1.1.1) <span class="citation">(Wickham, Vaughan, and Girlich
2023)</span> and purrr (v1.0.1) <span class="citation">(Wickham and
Henry 2023)</span>. The analysis project was managed using the workflowr
(v1.7.0) <span class="citation">(Blischak, Carbonetto, and Stephens
2021)</span> package which was also used to produce the publicly
available website displaying the analysis code, results and output.
Reproducible reports were produced using knitr (v1.42) <span
class="citation">(Xie 2023)</span> and R Markdown (v2.21) <span
class="citation">(Allaire et al. 2023)</span> and converted to HTML
using Pandoc (v).</p>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<div id="output-files" class="section level2">
<h2>Output files</h2>
<pre class="r"><code>versions &lt;- purrr::map(versions, as.character)
versions &lt;- jsonlite::toJSON(versions, pretty = TRUE)
readr::write_lines(versions,
                   here::here(&quot;3_Results&quot;, DOCNAME, &quot;package-versions.json&quot;))</code></pre>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-R-rmarkdown" class="csl-entry">
Allaire, JJ, Yihui Xie, Christophe Dervieux, Jonathan McPherson, Javier
Luraschi, Kevin Ushey, Aron Atkins, et al. 2023. <em>Rmarkdown: Dynamic
Documents for r</em>. <a
href="https://CRAN.R-project.org/package=rmarkdown">https://CRAN.R-project.org/package=rmarkdown</a>.
</div>
<div id="ref-R-workflowr" class="csl-entry">
Blischak, John, Peter Carbonetto, and Matthew Stephens. 2021.
<em>Workflowr: A Framework for Reproducible and Collaborative Data
Science</em>. <a
href="https://github.com/workflowr/workflowr">https://github.com/workflowr/workflowr</a>.
</div>
<div id="ref-R-ggasym" class="csl-entry">
Cook, Joshua H. 2021. <em>Ggasym: Asymmetric Matrix Plotting in
Ggplot2</em>. <a href="https://github.com/jhrcook/ggasym
https://jhrcook.github.io/ggasym/">https://github.com/jhrcook/ggasym
https://jhrcook.github.io/ggasym/</a>.
</div>
<div id="ref-damasMitoBreakMitochondrialDNA2014" class="csl-entry">
Damas, Joana, João Carneiro, António Amorim, and Filipe Pereira. 2014.
<span>“MitoBreak: the mitochondrial DNA breakpoints database.”</span>
<em>Nucleic Acids Research</em> 42 (Database issue): D1261–1268. <a
href="https://doi.org/10.1093/nar/gkt982">https://doi.org/10.1093/nar/gkt982</a>.
</div>
<div id="ref-freyClusteringPassingMessages2007" class="csl-entry">
Frey, Brendan J., and Delbert Dueck. 2007. <span>“Clustering by passing
messages between data points.”</span> <em>Science (New York, N.Y.)</em>
315 (5814): 972–76. <a
href="https://doi.org/10.1126/science.1136800">https://doi.org/10.1126/science.1136800</a>.
</div>
<div id="ref-guoRepeatsLongevitySources2010" class="csl-entry">
Guo, Xinhong, Konstantin Yu Popadin, Natalya Markuzon, Yuriy L. Orlov,
Yevgenya Kraytsberg, Kim J. Krishnan, Gábor Zsurka, Douglas M. Turnbull,
Wolfram S. Kunz, and Konstantin Khrapko. 2010. <span>“Repeats, longevity
and the sources of mtDNA deletions: evidence from ’deletional
spectra’.”</span> <em>Trends in genetics: TIG</em> 26 (8): 340–43. <a
href="https://doi.org/10.1016/j.tig.2010.05.006">https://doi.org/10.1016/j.tig.2010.05.006</a>.
</div>
<div id="ref-R-ggpubr" class="csl-entry">
Kassambara, Alboukadel. 2023. <em>Ggpubr: Ggplot2 Based Publication
Ready Plots</em>. <a
href="https://rpkgs.datanovia.com/ggpubr/">https://rpkgs.datanovia.com/ggpubr/</a>.
</div>
<div id="ref-lorenzViennaRNAPackage2011" class="csl-entry">
Lorenz, Ronny, Stephan H. Bernhart, Christian Höner Zu Siederdissen,
Hakim Tafer, Christoph Flamm, Peter F. Stadler, and Ivo L. Hofacker.
2011. <span>“ViennaRNA Package 2.0.”</span> <em>Algorithms for molecular
biology: AMB</em> 6 (November): 26. <a
href="https://doi.org/10.1186/1748-7188-6-26">https://doi.org/10.1186/1748-7188-6-26</a>.
</div>
<div id="ref-mcinnes2017" class="csl-entry">
McInnes, Leland, and John Healy. 2017. <span>“2017 IEEE International
Conference on Data Mining Workshops (ICDMW).”</span> In, 33–42. New
Orleans, LA: IEEE. <a
href="https://doi.org/10.1109/ICDMW.2017.12">https://doi.org/10.1109/ICDMW.2017.12</a>.
</div>
<div id="ref-needlemanGeneralMethodApplicable1970" class="csl-entry">
Needleman, S. B., and C. D. Wunsch. 1970. <span>“A general method
applicable to the search for similarities in the amino acid sequence of
two proteins.”</span> <em>Journal of Molecular Biology</em> 48 (3):
443–53. <a
href="https://doi.org/10.1016/0022-2836(70)90057-4">https://doi.org/10.1016/0022-2836(70)90057-4</a>.
</div>
<div id="ref-R-patchwork" class="csl-entry">
Pedersen, Thomas Lin. 2022. <em>Patchwork: The Composer of Plots</em>.
<a
href="https://CRAN.R-project.org/package=patchwork">https://CRAN.R-project.org/package=patchwork</a>.
</div>
<div id="ref-perssonCopychoiceRecombinationMitochondrial2019a"
class="csl-entry">
Persson, Örjan, Yazh Muthukumar, Swaraj Basu, Louise Jenninger, Jay P.
Uhler, Anna-Karin Berglund, Robert McFarland, et al. 2019.
<span>“Copy-choice recombination during mitochondrial L-strand synthesis
causes DNA deletions.”</span> <em>Nature Communications</em> 10 (1):
759. <a
href="https://doi.org/10.1038/s41467-019-08673-5">https://doi.org/10.1038/s41467-019-08673-5</a>.
</div>
<div id="ref-rao3DMapHuman2014" class="csl-entry">
Rao, Suhas S. P., Miriam H. Huntley, Neva C. Durand, Elena K. Stamenova,
Ivan D. Bochkov, James T. Robinson, Adrian L. Sanborn, et al. 2014.
<span>“A 3D map of the human genome at kilobase resolution reveals
principles of chromatin looping.”</span> <em>Cell</em> 159 (7): 1665–80.
<a
href="https://doi.org/10.1016/j.cell.2014.11.021">https://doi.org/10.1016/j.cell.2014.11.021</a>.
</div>
<div id="ref-robinsonJuiceboxJsProvides2018" class="csl-entry">
Robinson, James T., Douglass Turner, Neva C. Durand, Helga
Thorvaldsdóttir, Jill P. Mesirov, and Erez Lieberman Aiden. 2018.
<span>“Juicebox.js Provides a Cloud-Based Visualization System for Hi-C
Data.”</span> <em>Cell Systems</em> 6 (2): 256–258.e1. <a
href="https://doi.org/10.1016/j.cels.2018.01.001">https://doi.org/10.1016/j.cels.2018.01.001</a>.
</div>
<div id="ref-shamanskiyImtRDBDatabaseSoftware2019" class="csl-entry">
Shamanskiy, Viktor A., Valeria N. Timonina, Konstantin Yu Popadin, and
Konstantin V. Gunbin. 2019. <span>“ImtRDB: a database and software for
mitochondrial imperfect interspersed repeats annotation.”</span> <em>BMC
genomics</em> 20 (Suppl 3): 295. <a
href="https://doi.org/10.1186/s12864-019-5536-1">https://doi.org/10.1186/s12864-019-5536-1</a>.
</div>
<div id="ref-R-tidyverse" class="csl-entry">
Wickham, Hadley. 2023. <em>Tidyverse: Easily Install and Load the
Tidyverse</em>. <a
href="https://CRAN.R-project.org/package=tidyverse">https://CRAN.R-project.org/package=tidyverse</a>.
</div>
<div id="ref-R-dplyr" class="csl-entry">
Wickham, Hadley, Romain François, Lionel Henry, Kirill Müller, and Davis
Vaughan. 2023. <em>Dplyr: A Grammar of Data Manipulation</em>. <a
href="https://CRAN.R-project.org/package=dplyr">https://CRAN.R-project.org/package=dplyr</a>.
</div>
<div id="ref-R-purrr" class="csl-entry">
Wickham, Hadley, and Lionel Henry. 2023. <em>Purrr: Functional
Programming Tools</em>. <a
href="https://CRAN.R-project.org/package=purrr">https://CRAN.R-project.org/package=purrr</a>.
</div>
<div id="ref-R-tidyr" class="csl-entry">
Wickham, Hadley, Davis Vaughan, and Maximilian Girlich. 2023. <em>Tidyr:
Tidy Messy Data</em>. <a
href="https://CRAN.R-project.org/package=tidyr">https://CRAN.R-project.org/package=tidyr</a>.
</div>
<div id="ref-R-cowplot" class="csl-entry">
Wilke, Claus O. 2020. <em>Cowplot: Streamlined Plot Theme and Plot
Annotations for Ggplot2</em>. <a
href="https://wilkelab.org/cowplot/">https://wilkelab.org/cowplot/</a>.
</div>
<div id="ref-R-knitr" class="csl-entry">
Xie, Yihui. 2023. <em>Knitr: A General-Purpose Package for Dynamic
Report Generation in r</em>. <a
href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
<div id="ref-zazhytskaNoncellautonomousDisruptionNuclear2022"
class="csl-entry">
Zazhytska, Marianna, Albana Kodra, Daisy A. Hoagland, Justin Frere, John
F. Fullard, Hani Shayya, Natalie G. McArthur, et al. 2022.
<span>“Non-cell-autonomous disruption of nuclear architecture as a
potential cause of COVID-19-induced anosmia.”</span> <em>Cell</em> 185
(6): 1052–1064.e12. <a
href="https://doi.org/10.1016/j.cell.2022.01.024">https://doi.org/10.1016/j.cell.2022.01.024</a>.
</div>
<div id="ref-zukerMfoldWebServer2003" class="csl-entry">
Zuker, Michael. 2003. <span>“Mfold web server for nucleic acid folding
and hybridization prediction.”</span> <em>Nucleic Acids Research</em> 31
(13): 3406–15. <a
href="https://doi.org/10.1093/nar/gkg595">https://doi.org/10.1093/nar/gkg595</a>.
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] lubridate_1.9.2 forcats_1.0.0   stringr_1.5.0   dplyr_1.1.1    
 [5] purrr_1.0.1     readr_2.1.4     tidyr_1.3.0     tibble_3.2.1   
 [9] ggplot2_3.4.2   tidyverse_2.0.0 jsonlite_1.8.4  knitr_1.42     
[13] glue_1.6.2      workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0 xfun_0.38        bslib_0.4.2      colorspace_2.1-0
 [5] vctrs_0.6.1      generics_0.1.3   htmltools_0.5.5  yaml_2.3.7      
 [9] utf8_1.2.3       rlang_1.1.0      jquerylib_0.1.4  later_1.3.0     
[13] pillar_1.9.0     withr_2.5.0      bit64_4.0.5      lifecycle_1.0.3 
[17] munsell_0.5.0    gtable_0.3.3     evaluate_0.20    tzdb_0.3.0      
[21] callr_3.7.3      fastmap_1.1.1    httpuv_1.6.9     ps_1.7.4        
[25] parallel_4.2.2   fansi_1.0.4      Rcpp_1.0.10      renv_0.17.2     
[29] promises_1.2.0.1 scales_1.2.1     cachem_1.0.7     vroom_1.6.1     
[33] bit_4.0.5        fs_1.6.1         hms_1.1.3        digest_0.6.31   
[37] stringi_1.7.12   processx_3.8.0   getPass_0.2-2    rprojroot_2.0.3 
[41] grid_4.2.2       here_1.0.1       cli_3.6.1        tools_4.2.2     
[45] magrittr_2.0.3   sass_0.4.5       crayon_1.5.2     whisker_0.4.1   
[49] pkgconfig_2.0.3  timechange_0.2.0 rmarkdown_2.21   httr_1.4.5      
[53] rstudioapi_0.14  R6_2.5.1         git2r_0.31.0     compiler_4.2.2  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
